llama-index-core/llama_index/core/base/llms/types.py

```diff
@@ -85,12 +85,14 @@ def image_to_base64(self) -> Self:
             # Not base64 - encode it
             self.image = base64.b64encode(self.image)
 
+        self._guess_mimetype(decoded_img)
+        return self
+
+    def _guess_mimetype(self, img_data: bytes) -> None:
```

Comment:
Adding this function to guess the mimetype not only in the init but also every time we resolve the image